internal/cmd/plugin/command_install.go

```diff
@@ -0,0 +1,232 @@
+		if err := yaml.Unmarshal(manifestFile, manifest); err != nil {
+			return nil, err
+		}
+		manifest.Name = file.Name()
+
+		return manifest, nil
+	}
+
+	return nil, errors.Errorf("plugin %s not found", pluginName)
+}
+
+func installPlugin(manifest *Manifest, repositoryDir, installDir string) error {
+	language, ver := getLanguage(manifest)
+
+	switch language {
+	case Go:
+		checkGoVersion(ver)
+		return installGoPlugin(manifest.Name)
+	case Python:
+		checkPythonVersion(ver)
+		return installSimplePlugin(manifest.Name, repositoryDir, installDir, Python)
+	case Shell:
+		return installSimplePlugin(manifest.Name, repositoryDir, installDir, Shell)
+	default:
+		return errors.Errorf("installation of plugins using %s is not yet supported", language)
+	}
+}
+
+func getLanguage(manifest *Manifest) (string, *version.Version) {
+	if manifest == nil || len(manifest.Dependencies) == 0 {
+		return "", nil
+	}
+
+	primaryDependency := manifest.Dependencies[0]
+	primaryDependency.Dependency = strings.ToLower(primaryDependency.Dependency)
+	if primaryDependency.Version == "" {
+		return primaryDependency.Dependency, nil
+	}
+
+	ver, err := version.NewVersion(primaryDependency.Version)
+	if err != nil {
+		return primaryDependency.Dependency, nil
+	}
+
+	return primaryDependency.Dependency, ver
+}
+
+func checkPythonVersion(ver *version.Version) {
+	versionCmd := exec.NewCommand(Python, "--version")
+
+	out, err := versionCmd.Output()
+	if err != nil {
+		output.ErrPrintf(programNotFoundMsg, Python)
+		return
+	}
+
+	re := regexp.MustCompile(`^[1-9][0-9]*\.[0-9]+\.(0|[1-9][0-9]*)$`)
+	for _, word := range strings.Split(string(out), " ") {
+		if re.MatchString(word) {
+			installedVer, err := version.NewVersion(strings.Trim(word, " \n"))
+			if err != nil {
+				output.ErrPrintf(unableToParseVersionMsg, Python)
+				return
+			}
+			if installedVer.LessThan(ver) {
+				output.ErrPrintf(insufficientVersionMsg, Python, installedVer, ver)
+				return
+			}
+		}
+	}
+}
+
+func checkGoVersion(ver *version.Version) {
+	versionCmd := exec.NewCommand(Go, "version")
+
+	out, err := versionCmd.Output()
+	if err != nil {
+		output.ErrPrintf(programNotFoundMsg, Go)
+		return
+	}
+
+	re := regexp.MustCompile(`^go[1-9][0-9]*\.[0-9]+(\.[1-9][0-9]*)?$`)
+	for _, word := range strings.Split(string(out), " ") {
+		if re.MatchString(word) {
+			installedVer, err := version.NewVersion(strings.TrimPrefix(word, "go"))
+			if err != nil {
+				output.ErrPrintf(unableToParseVersionMsg, Go)
+				return
+			}
+			if installedVer.LessThan(ver) {
+				output.ErrPrintf(insufficientVersionMsg, Go, installedVer, ver)
+				return
+			}
+		}
+	}
+}
+
+func installSimplePlugin(name, repositoryDir, installDir, language string) error {
+	pluginDir := fmt.Sprintf("%s/%s", repositoryDir, name)
+	files, err := os.ReadDir(pluginDir)
+	if err != nil {
+		return err
+	}
+
+	found := false
+	for _, file := range files {
+		if strings.HasPrefix(file.Name(), "confluent-") {
+			found = true
```

Comment:
Hmm, this code should already exist for `confluent plugin list`. Wonder if we can repurpose it in some way.